file/writable_file_writer.h

```diff
@@ -234,11 +234,16 @@ class WritableFileWriter {
 
   // When this Append API is called, if the crc32c_checksum is not provided, we
   // will calculate the checksum internally.
-  IOStatus Append(const Slice& data, uint32_t crc32c_checksum = 0);
+  IOStatus Append(const Slice& data, uint32_t crc32c_checksum = 0,
+                  Env::IOPriority op_rate_limiter_priority = Env::IO_TOTAL,
+                  bool op_override_file_priority = false);
```

Comment:
Can a provided op-level priority (i.e., non-`Env::IO_TOTAL`) always override the file-level priority? It feels expected that the finer granularity setting overrides.

Reply:
> Can a provided op-level priority (i.e., non-Env::IO_TOTAL) always override the file-level priority? 

Yes, I believe this is how `WritableFileWriter::DecideRateLimiterPriority` works except that it is written more verbosely (and I can change that).

But the issue `op_override_file_priority` was trying to solve is when `op-level priority == Env::IO_TOTAL`, where it does not override file-level-pri for compact/flush and "does" override for WAL (more for a future hypothetical case where "what-if" WAL's io priority is set to non-Env::IO_TOTAL and developers forget to test the compatibility of rate-limiting auto WAL flush and non-Env::IO_TOTAL WAL's io pri ). 

But I am also open for discussion whether it worths the extra parameter for the "hypothetical case".

Reply:
> future hypothetical case where "what-if" WAL's io priority is set to non-Env::IO_TOTAL and developers forget to test the compatibility of rate-limiting auto WAL flush and non-Env::IO_TOTAL WAL's io pri ).

non-Env::IO_TOTAL in the WAL and rate-limiting auto WAL flush means the auto WAL flush priority should override. And it does because auto-flush callers set this value to true. Whereas, other WAL users leave this as false - but they also don't pass a op-level priority, so it'd take the file-level priority regardless of whether this is false/true. So actually I still don't really understand the problem we're solving.

Reply:
Oh I see, this doesn't behave the way I thought it would. Setting this flag forces op-level priority to override in every case. OK let me rethink it...

Reply:
Here is my new understanding: we set this flag to true in auto-flushes, then somebody who comes in the future and gives a file-level priority WAL feature must notice there are op-level priorities too because their new feature won't work at all. Then they make sure it's compatible and set it to false. Well, the logic makes sense now. I wouldn't personally add code complexity to give hypothetical feature implementors a smooth path to not learning the surrounding code, but maybe I'm just mean. So will leave it up to you.

Reply:
Discussed offline and concluded that we can clarify FSWritableFile::SetIOPriority and addressed the concern of mine behind adding the extra parameter "op_override_file_priority"

Fixed.

Reply:
This would've been extra difficult if we had continued trying to keep #9606 separate

Reply:
TIL!